Update item count badge styles

Aljullu opened the pull request:
Fixes #2617.

Small PR updating the item count badge styles in the _Cart_ block.

### Screenshots
| Before | After |
| --- | --- |
| ![imatge](https://user-images.githubusercontent.com/3616980/83643144-f6ff2e80-a5af-11ea-9df5-8e68ef01dfa3.png) | ![imatge](https://user-images.githubusercontent.com/3616980/83643024-d9ca6000-a5af-11ea-85a3-c98bb1a4e00a.png) |

### How to test the changes in this Pull Request:
Open the _Cart_ block and verify the item count badge matches the new styles.

### Changelog

> The item count badges of the Checkout block have been updated so it looks better in light & dark backgrounds.

nerrad commented:
Twenty Twenty styling looks a little bit awkward (padding is off a bit), so you may want to see how it looks with larger numbers?

<img width="442" alt="Image 2020-06-03 at 10 36 56 AM" src="https://user-images.githubusercontent.com/1429108/83650345-3d3f9800-a586-11ea-864a-e664c1a8d4da.png">

senadir commented:
Totally subjective but if we’re hardcoding colors why not a white circle and black text?

Aljullu commented:
> Twenty Twenty styling looks a little bit awkward (padding is off a bit)

I'm unable to reproduce, this is how it looks for me in 2020: :confused: 
![imatge](https://user-images.githubusercontent.com/3616980/83659880-6848dc80-a5c4-11ea-9541-0ae47b3ed241.png)

In your screenshot, the Panel arrow looks misaligned too. Maybe you switched branches and styles were not correctly generated?

> so you may want to see how it looks with larger numbers?

I initially tested with `99`, which is the largest number of items I could add with the _Cart_ block and it was working fine. In 801181ee2361b1909bd2f4e583c93127a5ab7257 I just added `white-space: nowrap;` to make sure it never breaks in two lines whatever the number is:
![imatge](https://user-images.githubusercontent.com/3616980/83660150-c70e5600-a5c4-11ea-997e-35701fb965ee.png)

> I must confess I like the original style better (completely subjective) but I understand the reasoning behind this change. It just feels so heavy compared with the other elements on the page (for storefront at least).

I see what you mean, even though in the previous design I thought the other way around, it felt the count badges were easy to miss.

One option might be to try to retrieve the background color via JS and then make the count badge match the theme colors. But I'm a bit wary of this because background colors might have transparencies, be gradients, images, etc. So it would not be bullet-proof. @nerrad what do you think?

Another option would be to leave this PR as-is, and then expect themes to customize these colors. We could take care of updating Storefront and we could add docs for 3rd-party themes.

PD: that's an example of an issue that we would not have with global styles. :slightly_smiling_face: 

> Totally subjective but if we’re hardcoding colors why not a white circle and black text?

For reference, this is how it would look:
![imatge](https://user-images.githubusercontent.com/3616980/83661212-1608bb00-a5c6-11ea-9875-1fd165d6712c.png)

cc'ing @LevinMedia in case he has some feedback on this.

LevinMedia commented:
@Aljullu @senadir 

>Totally subjective but if we’re hardcoding colors why not a white circle and black text?

Sorry I missed this option when I was doing the original designs, but yeah, that looks much better. Let's go with that.

Aljullu commented:
Summary of the last commits:
* 6b0ab17e4f22da6b731c1f0419eda730b336d99a switches black and white colors following the feedback above. :point_up_2: 
* b926fd89efa61b9e9b58f58d346d0d4cd35c6da5 adds documentation for themes on how to change the colors of the count badge.
* 6a047126c8ac2a53576f295412d76c87e817c021 moves _Order summary_ CSS selectors out of the parent class `.wc-block-checkout__sidebar`. This shouldn't make any visual difference but will make it easier for themes to style it because specificity is lower so a single class is enough to modify the styles.

nerrad commented:
I definitely like the badge designs better. In testing I discovered some more things and I realize you've merged and closed this pull (which is okay) but if there aren't known issues for these will you see that issues get created? I also know it's likely at least two of these issues weren't existing from the work in this pull, but I just discovered them so thought I'd make note of them 😄 

This screenshot has inline comments (comments apply to Storefront and Twenty Twenty):

<img width="1150" alt="Image 2020-06-05 at 11 03 32 AM" src="https://user-images.githubusercontent.com/1429108/83892975-5cbcf900-a71d-11ea-8f94-82bb9c7d9365.png">

The padding and margins on the validation for credit card fields seems off:

<img width="736" alt="Image 2020-06-05 at 11 07 50 AM" src="https://user-images.githubusercontent.com/1429108/83893111-88d87a00-a71d-11ea-98e9-d1d762c90b96.png">

I still see some off center padding for the numbers in the quantity bubbles for Twenty Twenty (both frontend and editor/mobile):

<img width="444" alt="Image 2020-06-05 at 11 10 32 AM" src="https://user-images.githubusercontent.com/1429108/83893231-b1f90a80-a71d-11ea-9d77-c22801f6396b.png">

For the above, I noticed you mentioned you couldn't reproduce, what's odd is that the font in your screenshot is much different than what's in my screenshot for the Twenty Twenty Theme. Since I don't see a place in the theme settings for customizing font, I'm wondering if this is a case of your system font being different than the system font displayed on my OS (and Twenty Twenty is defaulting to system font for this style)?

Aljullu commented:
Ah, sorry! I went ahead merging because it was already approved and there wasn't any update. I should have asked for another review instead. :sweat_smile: 

> This screenshot has inline comments (comments apply to Storefront and Twenty Twenty):

1a. Express checkout left/right padding: I'm unable to reproduce. Is that with Storefront `master` and WC Blocks `master`?
1b. Express checkout vertical align: right, good eye with that! I fixed it in: #2652.
2. The title vertical centering off will be fixed with #2649.

> The padding and margins on the validation for credit card fields seems off:

Created an issue: #2650. I added it to 2.7.0 milestone because I expect it to be a quick fix, but might move to 2.8.0 if I see it's harder to fix, I don't think it should be blocking.

> I still see some off center padding for the numbers in the quantity bubbles for Twenty Twenty (both frontend and editor/mobile):

Yeah, still unable to reproduce it here. I created an issue to investigate it: #2651. This one targeting 2.8.0.

nerrad commented:
> Ah, sorry! I went ahead merging because it was already approved and there wasn't any update. I should have asked for another review instead.

No worries 😄 ! I think it was fine to merge, we can handle the others in followups.

> 1a. Express checkout left/right padding: I'm unable to reproduce. Is that with Storefront master and WC Blocks master?

Storefront 2.5.7 and WC Blocks this branch.

Aljullu commented:
> > 1a. Express checkout left/right padding: I'm unable to reproduce. Is that with Storefront master and WC Blocks master?
> 
> Storefront 2.5.7 (released) and WC Blocks this branch.

I was still unable to reproduce this issue. But #2704 refactors the Express checkout CSS, so it might be a good chance to test if it's fixed now.

nerrad commented:
Still not seeing much padding on #2704 branch:

<img width="748" alt="Image 2020-06-12 at 11 27 04 AM" src="https://user-images.githubusercontent.com/1429108/84519319-e9c4fc80-ac9f-11ea-9bbd-583b78a97738.png">

(this is a production build)
